internals/overlord/logstate/clienterr/clienterr.go

```diff
@@ -16,6 +16,14 @@
 Package clienterr contains common error types which are recognised by the log
 gatherer. logstate.logClient implementations should return these error types
 to communicate with the gatherer.
+
+Errors in this package should be pattern-matched using errors.As:
+
+		err := client.Flush(ctx)
+		backoff := &clienterr.Backoff{}
```

Comment:
Note that you don't need to allocate a new instance (it'll get overwritten), just create a variable of that type. So you'd normally just do `var backoff *clienterr.Backoff`. See also the `As` examples in the package docs: https://pkg.go.dev/errors#As

Reply:
Good catch